Update issue templates (#1064)
aws/eks-distro: .github/ISSUE_TEMPLATE/release-tracking-issue.md

---
name: Release Tracking Issue
about: Handy template for release tracking
title: 'Releases: '
labels: release
assignees: ''

---

## KEY

> **✏️ –– PR opened**

Reviewer: ```suggestion
✏️ –– PR opened
`1-XX` –– used a placeholder for release branch (e.g., `1-23`) in inline code formatted text
`Z` –– used a placeholder for release number (e.g., `3`) in inline code formatted text
```

---
## {RELEASE_VERSION_NUMBER}

> - [ ]  Check for new project versions

Reviewer: ```suggestion
### Handle Existing PRs and Start Code Freeze for Non-Release Related Changes
- [ ]  For PRs created by `eks-distro-pr-bot` in [eks-distro](https://github.com/aws/eks-distro/pulls/eks-distro-pr-bot), [eks-distro-build-tooling](https://github.com/aws/eks-distro-build-tooling/pulls/eks-distro-pr-bot), and [eks-distro-prow-jobs](https://github.com/aws/eks-distro-prow-jobs/pulls/eks-distro-pr-bot), review and merge (when appropriate)
- [ ]  For non-`eks-distro-pr-bot` PRs in [eks-distro](https://github.com/aws/eks-distro/pulls?q=is%3Apr+is%3Aopen+-author%3Aeks-distro-pr-bot), [eks-distro-build-tooling](https://github.com/aws/eks-distro-build-tooling/pulls?q=is%3Apr+is%3Aopen+-author%3Aeks-distro-pr-bot), and [eks-distro-prow-jobs](https://github.com/aws/eks-distro-prow-jobs/pulls?q=is%3Apr+is%3Aopen+-author%3Aeks-distro-pr-bot), review and merge all un-held/non-WIP, production-ready PRs clearly intended for the release or release process; follow up with the authors of the remaining PRs about if the changes are needed for the release; work with the authors whose PRs are needed for the release get their proposed changes finished, reviewed and merged.
Release Changes
### Check for and Make Additional Release-Related Changes
- [ ]  Check for new project versions
```

  - [ ] kubernetes/kubernetes
  - [ ] containernetworking/plugins
  - [ ] coredns/coredns
  - [ ] kubernetes-sigs/metrics-server

>   - [ ] kubernetes-sigs/aws-iam-authenticator

Reviewer: ```suggestion
  - [ ] [kubernetes-sigs/aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/releases)
```

>   - [ ] kubernetes-sigs/metrics-server
>   - [ ] kubernetes-sigs/aws-iam-authenticator

Reviewer: ```suggestion
  - [ ] [kubernetes-sigs/metrics-server](https://github.com/kubernetes-sigs/metrics-server/releases)
  - [ ] [kubernetes-sigs/aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/releases)
```

  - [ ] etcd-io/etcd

>   - [ ] kubernetes-csi/external-snapshotter
>   - [ ] kubernetes-csi/external-attacher
>   - [ ] kubernetes-csi/external-provisioner
>   - [ ] kubernetes-csi/external-resizer
>   - [ ] kubernetes-csi/livenessprobe
>   - [ ] kubernetes-csi/node-driver-registrar

Reviewer: ```suggestion
  - [ ] [kubernetes-csi/external-attacher](https://github.com/kubernetes-csi/external-attacher/releases)
  - [ ] [kubernetes-csi/external-provisioner](https://github.com/kubernetes-csi/external-provisioner/releases)
  - [ ] [kubernetes-csi/external-resizer](https://github.com/kubernetes-csi/external-resizer/releases)
  - [ ] [kubernetes-csi/external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter/releases)
  - [ ] [kubernetes-csi/livenessprobe](https://github.com/kubernetes-csi/livenessprobe/releases)
  - [ ] [kubernetes-csi/node-driver-registrar](https://github.com/kubernetes-csi/node-driver-registrar/releases)
```

- [ ] Check for patch changes

> - [ ] Merge bot's attribution PR

Reviewer: ```suggestion
```

Reviewer: This is something we should just do and isn't specific to the release per se. I called it out last time because Jackson wanted this before the release, and we were waiting on it.

> - [ ] Cut dev release
> - [ ] Cut prod release
> - [ ] Confirm release manifest looks right
> - [ ] Confirm images are in ECR 

Reviewer: ```suggestion
### Cut Release and Confirm Success
- [ ] Cut dev release
  - [ ] Confirm `dev-release-1-XX-postsubmit` and `build-1-XX-postsubmit` [jobs](https://prow.eks.amazonaws.com/?repo=aws%2Feks-distro&type=postsubmit&job=*1-2*) finish successfully
- [ ] Cut prod release
  - [ ] Confirm `prod-release-1-XX-postsubmit` and `build-1-XX-postsubmit` [jobs](https://prow.eks.amazonaws.com/?repo=aws%2Feks-distro&type=postsubmit&job=*1-2*) finish successfully
- [ ] Confirm release manifest is available and looks right (`https://distro.eks.amazonaws.com/kubernetes-1-2X/kubernetes-1-2X-eks-#.yaml` <– replace `1-2X` with release branch and `#` with release number)
- [ ] Confirm images are in [ECR](https://gallery.ecr.aws/eks-distro) 
```

> - [ ] Update docs

Reviewer: ```suggestion
- [ ] Update docs  -- 1.19 - 1.22 only
```

>   - [ ] Include notification about updating asap

Reviewer: ```suggestion
```

Reviewer: This is 1-19 only and is temporary

- [ ] Tag release

> - [ ] Notify Slack channel

Reviewer: ```suggestion
- [ ] Generate Release on GitHub from tag (`RELEASE_BRANCH=1-XX make github-release`)
- [ ] Notify Slack channel
```

> - [ ] Update EKS-A 

Reviewer: ```suggestion
- [ ] Update EKS-A  (example [PR](https://github.com/aws/eks-anywhere-build-tooling/pull/965))
```

> - [ ] Cut dev release
> - [ ] Cut prod release
> - [ ] Confirm release manifest looks right
> - [ ] Confirm images are in ECR 
> - [ ] Update docs
>   - [ ] Include notification about updating asap
> - [ ] Tag release
> - [ ] Notify Slack channel
> - [ ] Update EKS-A 

Reviewer: ```suggestion
### Cut Release
- [ ] Open and merge dev release PR
- [ ] Confirm dev release was successful
  - [ ] Confirm `dev-release-1-XX-postsubmit` and `build-1-XX-postsubmit` [jobs](https://prow.eks.amazonaws.com/?repo=aws%2Feks-distro&type=postsubmit&job=*1-2*) finished successfully
- [ ] Open and merge prod release PR
- [ ] Confirm prod release was successful
  - [ ] Confirm that `prod-release-1-XX-postsubmit` and `build-1-XX-postsubmit` [jobs](https://prow.eks.amazonaws.com/?repo=aws%2Feks-distro&type=postsubmit&job=*1-2*) finished successfully
  - [ ] Confirm release manifest is available and looks right (`https://distro.eks.amazonaws.com/kubernetes-1-XX/kubernetes-1-XX-eks-Z.yaml`)
  - [ ] Confirm images are in [ECR](https://gallery.ecr.aws/eks-distro) 
### Announce Release (*1.19 - 1.22 only*)
- [ ] Update docs (`RELEASE_BRANCH=1-XX make docs` + additional manual changes) and get someone to actually review PR
- [ ] Confirm docs changes
  - [ ] Check [website](https://distro.eks.amazonaws.com/) was updated
  - [ ] Check that SNS was received
- [ ] Tag release (`RELEASE_GIT_TAG=v1-XX-eks-# RELEASE_GIT_COMMIT_HASH=<docs_commit_hash> make tag)`
- [ ] Generate [Release on GitHub](https://github.com/aws/eks-distro/releases) from tag  (`RELEASE_BRANCH=1-XX make github-release`)
- [ ] Notify Slack channel
### Update Other Projects
- [ ] Open EKS-A PR (see [example](https://github.com/aws/eks-anywhere-build-tooling/pull/965))
- [ ] Open BottleRocket PR  –– 1.22 - 1.23 only (see [example](https://github.com/bottlerocket-os/bottlerocket/pull/2230))
```

> - [ ] Update BottleRocket

Reviewer: ```suggestion
- [ ] Update BottleRocket -- 1.22 - 1.23 only (example [PR](https://github.com/bottlerocket-os/bottlerocket/pull/2230))
```
